src/Configuration/Metadata/Driver/AttributeDriver/AttributeReader.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Hateoas\Configuration\Metadata\Driver\AttributeDriver;
+
+use Doctrine\Common\Annotations\Reader;
+use ReflectionClass;
+use ReflectionMethod;
+use ReflectionProperty;
+
+/**
+ * @see \JMS\Serializer\Metadata\Driver\AttributeDriver\AttributeReader
+ */
+class AttributeReader implements Reader
+{
+    /**
+     * @var Reader
+     */
+    private $reader;
+
+    public function __construct(Reader $reader)
```

Comment:
From the class name I would expect that this class is only about reading attributes.

An approach similar to https://github.com/schmittjoh/serializer/pull/1469 would be nice.
This way the annotation reader would become optional.

Reply:
To be honest, I didn't want to do something very different from how JMS was doing it back then. That PR was created after I created mine.

But I totally agree with that decoupling approach.

I believe it answers my concern in this comment https://github.com/willdurand/Hateoas/pull/326#issuecomment-1888073337